PizzaBuilder/Program.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Generic;
+namespace PizzaBuilder
+{
+    public interface IPizzaBuilder
+    {
+        void buildDough();
+        void buildSauce();
+        void buildTopping();
+        void reset();
+    }
+
+    public class Director
+    {
+        private IPizzaBuilder _pizzaBuilder;
+
+        public IPizzaBuilder PizzaBuilder
+        {
+            set
+            {
+                this._pizzaBuilder = value;
+            }
+        }
+
+        public void makePizza()
+        {
+            this._pizzaBuilder.buildDough();
+            this._pizzaBuilder.buildSauce();
+            this._pizzaBuilder.buildTopping();
+        }
+
+        public void makePizzaNoSauce()
+        {
+            this._pizzaBuilder.buildDough();
+            this._pizzaBuilder.buildTopping();
+        }
+    }
+
+    public class WhitePizzaBuilder : IPizzaBuilder
+    {
+        private WhitePizza _whitePizza = new WhitePizza();
+
+        public WhitePizzaBuilder()
+        {
+            this.reset();
+        }
+
+        public void buildDough()
+        {
+            this._whitePizza.Add("White Dough");
+        }
+        public void buildSauce()
+        {
+            this._whitePizza.Add("Sauce");
+        }
+        public void buildTopping()

```

Comment:
It would be nice if you specified some toppings like (Broccoli, Mushroom, onions) or what you want rather than just writing the word Topping.